app/controllers/passengers_controller.rb

```diff
@@ -0,0 +1,75 @@
+class PassengersController < ApplicationController
+
+  def index
+    @passengers = Passenger.all
+  end
+
+  def show
+    @passenger = Passenger.find_by(id: params[:id])
+    if @passenger.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def new
+    @passenger = Passenger.new
+    if params[:passenger_id].nil?
+      @trip = Trip.new
+    else
+      passenger = passenger.find_by(id: params[:id])
+      passenger.trips.new
+      return
+    end
+  end
+
+  def create
+    @passenger = Passenger.new(passenger_params)
+    if @passenger.save
+      redirect_to passenger_path(@passenger.id)
+      return
+    else
+      render :new
+      return
+    end
+  end
+
+  def edit
+    @passenger = Passenger.find_by(id: params[:id])
+    if @passenger.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def update
+    @passenger = Passenger.find_by(id: params[:id])
+    if @passenger.nil?
+      head :not_found
+      return
+    elsif @passenger.update(passenger_params)
+      redirect_to passenger_path
+      return
+      render :edit, status: :bad_request
+      return
+    end
```

Comment:
Missing an else case here.  This is causing the validation messages not to show. 

```suggestion
    elsif @passenger.update(passenger_params)
      redirect_to passenger_path
      return
    else
      render :edit, status: :bad_request
      return
    end
```